src/services/EdFi.DataManagementService.Api/Core/Middleware/ValidateDocumentMiddleware.cs

```diff
@@ -21,6 +22,32 @@ public async Task Execute(PipelineContext context, Func<Task> next)
         _logger.LogDebug("Entering ValidateDocumentMiddleware- {TraceId}", context.FrontendRequest.TraceId);
 
         var validatorContext = new ValidatorContext(context.ResourceSchema, context.FrontendRequest.Method);
+
+        // First check for basic JSON errors
+        var jsonErrors = _documentValidator.ValidateJsonFormat(context.FrontendRequest.Body);
+        if (jsonErrors != null && jsonErrors.Any())
+        {
+            SendFailureResponse(context, jsonErrors.ToArray());
+        }
+
+        // Evaluate schema
+        var evaluationResult = _documentValidator.Evaluate(context.FrontendRequest.Body, validatorContext);
+
+        if (
+            _documentValidator.PruneOverPostedData(
+                context.FrontendRequest.Body,
+                evaluationResult,
+                out JsonNode? prunedBody
+            )
+        )
+        {
+            // Used pruned body for the remainder of pipeline
+            context.FrontendRequest = context.FrontendRequest with
```

Comment:
That's a [new syntax](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/with-expression) I've not encountered. Nice trick.